src/NUnitFramework/framework/Internal/TypeHelper.cs

```diff
@@ -198,7 +198,7 @@ public static bool TryGetBestCommonType(Type? type1, Type? type2, [NotNullIfNotN
             if (type1.IsAssignableFrom(type2)) { bestCommonType = type1; return true; }
             if (type2.IsAssignableFrom(type1)) { bestCommonType = type2; return true; }
 
-            bestCommonType = null;
+            bestCommonType = typeof(object);
```

Comment:
This cannot be `null` if `type1` and `type2` are not null which could be the case. It was causing a compile error, so I looked at the calling code. If we ever get here and return `false`, we immediately set another value to the `out` parameter.

Reply:
Oh, I missed this when I annotated. My preferred fix would be to add `[MaybeNullWhen(false)]` to the parameter, if that works alongside NotNullIfNotNull, but this is also fine. Either way, if this method returns false, the out parameter should not be read by the caller.